parsing/notgram_ops.ml

```diff
@@ -28,18 +28,20 @@ let grammar_of_notation ntn =
   NotationMap.find ntn !notation_grammar_map
 
 let list_prefixes ntn =
-  let rec aux = function
+  let rec aux inrec = function
     | [] -> []
     | s :: symbols ->
-       let nt = match s with
-         | NonTerminal _ -> 1
-         | Terminal _ | SProdList _ | Break _ -> 0 in
-       ([s], nt) :: List.map (fun (l, k) -> s :: l, k + nt) (aux symbols) in
+       let nt, inrec = match s with
+         | NonTerminal _ -> (if inrec then 0 else 1), false
+         | Terminal ".." -> 0, true  (* that's horribly hackish but we plan to
```

Comment:
> For more abstraction, you can use ` Terminal s when String.equal s (Id.to_string Notation_ops.ldots_var)`.

Done

> I'd be curious also of how it would rebase on top of #19404.

I'll do the rebase when working on it